Feature/spring integration v2

wolfdale opened the pull request:
Spring Integration Work

- [x] Dependencies added.
- [x] Candlepin Commons|API|Checks migrated.
- [x] Candlepin Server 
- [ ] <strike>Candlepin Deploy script changes [Can be skipped]</strike> 
- [ ] Candlepin Liquibase changes
- [ ] Verification

wolfdale committed:
Candlepin Spring - Dependencies added

wolfdale committed:
Candlepin Spring - Commons

wolfdale committed:
Candlepin Spring - Server Part - 1

wolfdale committed:
Candlepin Spring - Server Part - 2

wolfdale committed:
Candlepin Spring - Server Part - 3

AmitMandliya commented on server/src/main/java/org/candlepin/hostedtest/HostedTestSubscriptionServiceAdapter.java:
```diff
@@ -11,7 +11,7 @@
  * granted to use or replicate Red Hat trademarks that are incorporated
  * in this software or its documentation.
- */
+ */ConsumerTypeRe
```
This must have been pasted by mistake and should be removed.

wolfdale replied:
Fixed.

wolfdale committed:
Candlepin Spring - Server Fixes

wolfdale committed:
Candlepin Spring - Server Fixes (Reasteasy fix, JSProvider fix) [Needs Refactoring]

Commit:
Candlepin migration from guice to spring

- removes the usage of unitOfWork from JobMessageReceiver
- defaults the I18nProvider to use the locale.US

Commit:
Candlein migration from guice to spring

- Removes the transitive dependency for PoolRules

wolfdale committed:
Refactoring + cleanup
